amoro-format-iceberg/src/main/java/org/apache/amoro/utils/TableFileUtil.java

```diff
@@ -199,7 +217,9 @@ public static String optimizingPosDeleteFileName(String dataFileName, String suf
   }
 
   public static boolean isOptimizingPosDeleteFile(String dataFilePath, String posDeleteFilePath) {
-    return getFileName(posDeleteFilePath)
-        .startsWith(String.format("%s-%s", getFileName(dataFilePath), POS_DELETE_FILE_IDENTIFIER));
+    return getFileNameWithoutExt(posDeleteFilePath)
```

Comment:
We may skip truncating the extent for the position delete file path here to reduce some cost as we determine by `startsWith`, HDYT?